homeassistant/components/opentherm_gw/sensor.py

```diff
@@ -1,20 +1,621 @@
+            native_unit_of_measurement=UnitOfTime.HOURS,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_CH_PUMP_HOURS,
+            friendly_name_format="Central Heating Pump Hours {}",
+            device_class=SensorDeviceClass.DURATION,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfTime.HOURS,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_DHW_PUMP_HOURS,
+            friendly_name_format="Hot Water Pump Hours {}",
+            device_class=SensorDeviceClass.DURATION,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfTime.HOURS,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_DHW_BURNER_HOURS,
+            friendly_name_format="Hot Water Burner Hours {}",
+            device_class=SensorDeviceClass.DURATION,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfTime.HOURS,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_MASTER_OT_VERSION,
+            friendly_name_format="Thermostat OpenTherm Version {}",
+            suggested_display_precision=SENSOR_FLOAT_SUGGESTED_DISPLAY_PRECISION,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_SLAVE_OT_VERSION,
+            friendly_name_format="Boiler OpenTherm Version {}",
+            suggested_display_precision=SENSOR_FLOAT_SUGGESTED_DISPLAY_PRECISION,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_MASTER_PRODUCT_TYPE,
+            friendly_name_format="Thermostat Product Type {}",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_MASTER_PRODUCT_VERSION,
+            friendly_name_format="Thermostat Product Version {}",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_SLAVE_PRODUCT_TYPE,
+            friendly_name_format="Boiler Product Type {}",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_SLAVE_PRODUCT_VERSION,
+            friendly_name_format="Boiler Product Version {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_MODE,
+            friendly_name_format="Gateway/Monitor Mode {}",
+        ),
+    ),
```

Comment:
What are the possible states

Reply:
`0` (Monitor mode), `1` (Gateway mode) or `R` (reset), the latter being a transient mode that is used to reset the OpenTherm Gateway.
I would have to dig deeper to see if this sensor could possibly have value `R`.

Reply:
I think this would be a good example for a refactor. So we have the `ENUM` device class. This sets up the sensor with a predefined list of options, which can be translated. Because the sensor has predefined options, we can use this in automations. So if you want to create an automation based on the state of an entity, we can just show a dropdown because we already know every possible state for example.

It would require a breaking change, as the value would go from `0` to `monitor_mode` (or `monitor`, whatever you prefer) and then we add the state translations in the `strings.json`. But since this would be breaking, please do this in a separate PR (I would recommend adding this after we sorted out device translations and potentially the split of devices)